src/Microsoft.Sbom.Api/FormatValidator/ValidatedSBOMFactory.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Sbom.Api.FormatValidator;
+
+using System.IO;
+
+public class ValidatedSBOMFactory
+{
+    public virtual IValidatedSBOM CreateValidatedSBOM(string sbomFilePath)
+    {
+        var sbomStream = new StreamReader(sbomFilePath);
```

Comment:
Does this need a `using` statement to properly dispose of the Stream?

Reply:
Hm, the lifespan of the stream is fuzzy here since both the caller and the ValidatedSBOM touch it. The ValidatedSBOM needs that stream to live at least long enough to deserialize it - which since it happens lazily, could be sometime after the ValidatedSBOM is created. So a using(stream) here could cause the stream to be disposed before the ValidatedSBOM has actually deserialized it.

Maybe the ValidatedSBOM should own the stream and responsibility for disposing it when it is disposed itself.

Reply:
Relying on the caller to provide cleanup of an `IDisposable` resource (in this case, the `Stream`) is less than ideal--in fact, [CA1001](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1001) is designed to avoid just this scenario.

I think any "proper" solution would involve deriving `IValidatedSBOM` from `IDisposable` and implementing the associated `Dispose` method. That may or may not be worth the small memory leak that is represented here, given the short program lifetime and the current time constraints.

If `ValidatedSBOM` implements `IDisposable`, then it might also make sense to revisit the constructor parameter. Instead of a `Stream`, it could represent a file path or a string that represents the file contents, and then the `Stream` object is fully contained inside `ValidatedSBOM`.

Reply:
Yeah, I agree that the caller should not need to know how long to keep the stream around and when it's ok to dispose it. The lifetime must be controlled by the only thing that has a stake in how long it lives, and that's the ValidatedSBOM. But, it's maybe a little unexpected that something else creates it, and hands off ownership.

I think we _don't_ want to be passing a string around because of the potential for very big SBOMs in memory. And I moved away from a file path in order to facilitate unit testing.

Maybe ValidatedSBOM should expose a Load(stream) method. Then the caller explicitly calls load, inside a using(stream) block. ValidatedSBOM no longer needs to store the stream at all, and the time when the load occurs is completely deterministic, controlled by the caller, and separate from object construction.